signer/src/storage/in_memory.rs

```diff
@@ -259,11 +260,66 @@ impl super::DbRead for SharedStore {
         let pending_deposit_requests = self
             .get_pending_deposit_requests(chain_tip, context_window)
             .await?;
-        let store = self.lock().await;
 
-        let threshold = threshold as usize;
+        let minimum_acceptable_reclaim_unlock_time = self
+            .get_bitcoin_block(chain_tip)
+            .await?
+            .map(|block| block.block_height as i32 + MINIMUM_RECLAIM_PROXIMITY_TO_CHAIN_TIP as i32)
+            .ok_or(Error::MissingBitcoinBlock(*chain_tip))?;
+
+        let chain_tip_ref = self
+            .get_bitcoin_block(chain_tip)
+            .await?
+            .map(|block| model::BitcoinBlockRef {
+                block_hash: *chain_tip,
+                block_height: block.block_height,
+            })
+            .unwrap();
```

Comment:
nit: call `self.get_bitcoin_block(chain_tip)` once and use the result to compute the two things?

Reply:
Changed the approach - effectively done.